packages/fuels-core/src/code_gen/abigen.rs

```diff
@@ -122,7 +128,48 @@ impl Abigen {
                            let provider = self.wallet.get_provider()?;
                            wallet.set_provider(provider.clone());
 
-                           Ok(Self { contract_id: self.contract_id.clone(), wallet: wallet })
+                           Ok(Self { contract_id: self.contract_id.clone(), wallet: wallet, logs_lookup: self.logs_lookup.clone() })
+                        }
+
+                        pub fn _logs_with_type<D: Tokenizable + Parameterize>(&self, receipts: &[Receipt]) -> Result<Vec<D>, SDKError> {
+                            let param_type = D::param_type();
+
+                            let target_ids: HashSet<u64> = self
+                                .logs_lookup
+                                .iter()
+                                .filter_map(|l| {
+                                    if l.param_type == param_type {
+                                        Some(l.log_id)
+                                    } else {
+                                        None
+                                    }
+                                })
+                                .collect();
+
+                            let decode_log_content = |content: &[u8]| {
```

Comment:
there is a function in `fuel-core` that does this. It is called `try_from_bytes`. Wherever this closure is called, that function can be called instead. It is already in the set of `use`s added to the `abigen`ed code.